api/v1/menus/create.js

```diff
@@ -28,16 +29,31 @@ module.exports = (req, res, next) => {
     let fsClient = new FSClient();
     let dbClient = new DBClient('menus');
     fsClient.uploadMenu(req.file)
-        .then(id => {
-            menuEntry.file = `/files/${id}`
-            return deleteFile(req.file.path);
+        .then((id) => {
+            menuEntry.file = `/files/${id}`;
+            return dbClient.insert(menuEntry)
+                .then((dbEntry) => {
```

Comment:
please, do not nest promises, it's an [anti-pattern](http://taoofcode.net/promise-anti-patterns/).

The best approach here would be something in the lines:

```js
fsClient.uploadMenu(req.file)
.then(id => {
    menuEntry.file = `/files/${id}`;
    return dbClient.insert(menuEntry);
)
.then(dbEntry => [...])
.catch(err => [...])
.then([OCR stuff])
```